tests/load/test_read_interfaces.py

```diff
@@ -0,0 +1,218 @@
+
+    # we always expect 2 chunks based on the above setup
+    expected_chunk_counts = [chunk_size, total_records - chunk_size]
+
+    @dlt.source()
+    def source():
+        @dlt.resource()
+        def items():
+            yield from [
+                {"id": i, "children": [{"id": i + 100}, {"id": i + 1000}]}
+                for i in range(total_records)
+            ]
+
+        return [items]
+
+    # run source
+    s = source()
+    pipeline.run(s, loader_file_format=destination_config.file_format)
+
+    # access via key
+    relationship = pipeline.dataset()["items"]
+
+    # full frame
+    df = relationship.df()
+    assert len(df.index) == total_records
+
+    #
+    # check dataframes
+    #
+
+    # chunk
+    df = relationship.df(chunk_size=chunk_size)
+    if not skip_df_chunk_size_check:
+        assert len(df.index) == chunk_size
+    assert set(df.columns.values) == set(expected_columns)
+
+    # iterate all dataframes
+    frames = list(relationship.iter_df(chunk_size=chunk_size))
+    if not skip_df_chunk_size_check:
+        assert [len(df.index) for df in frames] == expected_chunk_counts
+
+    # check all items are present
+    ids = reduce(lambda a, b: a + b, [f[expected_columns[0]].to_list() for f in frames])
+    assert set(ids) == set(range(total_records))
+
+    # access via prop
+    relationship = pipeline.dataset().items
+
+    #
+    # check arrow tables
+    #
+
+    # full table
+    table = relationship.arrow()
+    assert table.num_rows == total_records
+
+    # chunk
+    table = relationship.arrow(chunk_size=chunk_size)
+    assert set(table.column_names) == set(expected_columns)
+    assert table.num_rows == chunk_size
+
+    # check frame amount and items counts
+    tables = list(relationship.iter_arrow(chunk_size=chunk_size))
+    assert [t.num_rows for t in tables] == expected_chunk_counts
+
+    # check all items are present
+    ids = reduce(lambda a, b: a + b, [t.column(expected_columns[0]).to_pylist() for t in tables])
+    assert set(ids) == set(range(total_records))
+
+    # check fetch accessors
+    relationship = pipeline.dataset().items
+
+    # check accessing one item
+    one = relationship.fetchone()
+    assert one[0] in range(total_records)
+
+    # check fetchall
+    fall = relationship.fetchall()
+    assert len(fall) == total_records
+    assert {item[0] for item in fall} == set(range(total_records))
+
+    # check fetchmany
+    many = relationship.fetchmany(chunk_size)
+    assert len(many) == chunk_size
+
+    # check iterfetchmany
+    chunks = list(relationship.iter_fetchmany(chunk_size=chunk_size))
+    assert [len(chunk) for chunk in chunks] == expected_chunk_counts
+    ids = reduce(lambda a, b: a + b, [[item[0] for item in chunk] for chunk in chunks])
+    assert set(ids) == set(range(total_records))
+
+    # simple check that query also works
+    relationship = pipeline.dataset().query("select * from items where id < 20")
+
+    # we selected the first 20
```

Comment:
will need better tests

Reply:
actually probably not, this simple test should be enough to check that sql queries are passed on.